Develop

plumsoftware opened the pull request:
4-я лаба по Web

plumsoftware committed:
[+] added theme colors
[+] added clsx

plumsoftware committed:
[+] Created primary button
[~] added several new colors in theme

plumsoftware committed:
[+] created top nav text
[+] created top nav container

plumsoftware committed:
[+] created header
[~] clean up code
[+] created profile top nav

plumsoftware committed:
[+] created header row with action (consult site) and image
[+] added new asset (header asset)

plumsoftware committed:
[+] added our services

plumsoftware committed:
[+] added assets
[+] card
[+] added children to OurServices

plumsoftware committed:
[+] added outlined button
[+] added several new images for cards
[+] added font into index.html and index.css

plumsoftware committed:
[+] added divider
[+] added promo components
[+] added images

plumsoftware committed:
[~] all 'px' units replaced with '%' units

plumsoftware committed:
[+] added slider with states
[+] added costomer image

plumsoftware committed:
[~] replaces 'px' units in root`s padding with '%'

plumsoftware committed:
[+] added footer
[~] simple padding changes

plumsoftware committed:
[+] Добавил промо компонент

plumsoftware committed:
[+] simple code changes

konstantinzelinsky commented:
1. Проверка GitHub репозитория
Стандартное Readme
0 + 1 + 1 + 1
2. Запуск проекта
Проект не билдится
5
<img width="749" alt="image" src="https://github.com/plumsoftware/Lab4/assets/57330800/98eac523-fc3a-4cd5-a5d4-ad93bf1640b2">


3. Оценка прогресса выполнения
Сверстаны все разделы сайта, но есть отличия от макета, весьма весомые. Отступы, размеры, текст, не хватает бэкграунд элементов
7
4. Оценка верстки экрана
Шрифты не имеют такие же параметры, как в фигме (жирность и т.д)
Адаптивность не работает/работает с проблемами
0 + 0 + 1 + 2
5. Оценка кода
Код разбит на компоненты, но в некоторых местах как-то недостаточно, а где-то наоборот сильно маленькие компоненты. Хедер можно было вынести в отдельный компонент. Футер лежит в отдельном компоненте, но тег footer вне его и оборачивает его. Кнопки можно было объединить в один компонент и в зависимости от пропса рендерить с нужными стилями
В коде встречаются захардкоженные цвета, непосредственно в коде компонента. Такое стоило сделать, используя стили и применяя их в зависимости от условия
В css при этом используются переменные для цветов
6 + 7 + 2 + 1
6. Дополнительно
Слайдер меняет свой размер при изменении слайда
3 + 0

Изначально у меня был применен шрифт другой, не такой как в макете
<img width="452" alt="image" src="https://github.com/plumsoftware/Lab4/assets/57330800/47a36910-e81f-48e4-9b8a-22b652e0faed">

plumsoftware committed:
[~] исправил везде шрифт на Mulish
[~] скописовал все параметры шрифтов из фигмы и вставил в код
[~] футер теперь в ффутере и хэдер теперь в хэдере

plumsoftware committed:
[+] сделал так, что при нажатии на кнопку "показать больше" карточек с промо информацией становится больше (доп. задание)

plumsoftware committed:
[~] перенёс карточки в отдельным компонент, чтобы код был читабельнее

plumsoftware committed:
[+] добавил промо для гита

plumsoftware committed:
[~] причесал код, чтобы проект билдился

plumsoftware committed:
Update README.md

konstantinzelinsky commented:
Проект по-прежнему не билдится из-за проблем с нейнмингом файлов/папок. В импорте с маленькой буквы, а называется с большой
<img width="1225" alt="image" src="https://github.com/plumsoftware/Lab4/assets/57330800/e3c910b9-b219-4a42-babc-3f8f44a31877">

За Readme и доп задание баллы поставил
